script_to_db.py

```diff
@@ -1,86 +1,99 @@
-HOST=os.getenv("HOST")
-PORT=os.getenv("PORT")
-DATABASE=os.getenv("DATABASE")
-URL = os.getenv("URL")
+USER = os.getenv("USER")
+PASSWORD = os.getenv("PASSWORD")
+HOST = os.getenv("HOST")
+PORT = os.getenv("PORT")
+DATABASE = os.getenv("DATABASE")
+url = os.getenv("URL")
 ONA_USERNAME = os.getenv("ONA_USERNAME")
 ONA_PASSWORD = os.getenv("ONA_PASSWORD")
-ona_auth=(ONA_USERNAME,ONA_PASSWORD)
+ona_auth = (ONA_USERNAME, ONA_PASSWORD)
 
-def fetch_data():
-    response=requests.get(url=URL,auth=ona_auth)
+
+def fetch_data(page_url, page_auth):
+    '''Function to get response from ona api in json format'''
+    response = requests.get(url=page_url, auth=page_auth, timeout=10)
     return response.json()
-  
+
 
 def create_table():
-    commands=('CREATE TABLE IF NOT EXISTS test_2(_id int primary key NOT NULL,_submitted_by varchar NOT NULL,specific_time time);')
+    '''Command to create a table in the postgres database'''
+    commands = '''CREATE TABLE IF NOT EXISTS test_2(
+        _id int primary key NOT NULL,
+        _submitted_by varchar NOT NULL,
+        specific_time time);'''
     execute_statements(commands)
-    return 
 
 
 def insert_statements():
-    table_values=evaluate_json_values()
-    tupl_lst= [tuple (x) for x in table_values]
-    for x in tupl_lst:
-        insert_command=f'insert into test_2(_id,_submitted_by,specific_time) values(%s,%s,%s);'
-        execute_statements(insert_command,*x)
-    return 
+    '''Generating insert statement to populate table in postgres'''
+    tupl_lst = [tuple(x) for x in table_values]
+    for lst in tupl_lst:
+        insert_command = f'{"insert into test_2(_id,_submitted_by,specific_time)values(%s,%s,%s);"}'
+        execute_statements(insert_command, *lst)
 
 
-def connection_to_db(): 
+def connection_to_db():
+    '''Function to connect to the postgres database'''
     try:
-        connection=psycopg2.connect(user=USER,password=PASSWORD,host=HOST,port=PORT,database=DATABASE)
-    except Exception as error:
+        connection = psycopg2.connect(
+            user=USER,
+            password=PASSWORD,
+            host=HOST,
+            port=PORT,
+            database=DATABASE
+            )
+    except ValueError as error:
         print(error)
     return connection
 
 
-def execute_statements(sql_command,*placeholder_values):
+def execute_statements(sql_command, *placeholder_values):
+    '''Function to execute commands'''
     try:
-        connection=connection_to_db()
-        conn=connection.cursor()
+        connection = connection_to_db()
+        conn = connection.cursor()
         if placeholder_values is None:
             conn.execute(sql_command)
             connection.commit()
         if placeholder_values is not None:
-            conn.execute(sql_command,placeholder_values)
+            conn.execute(sql_command, placeholder_values)
             connection.commit()
         close_connection(connection)
-    except Exception as error:
+    except ValueError as error:
```

Comment:
The `ValueError` exception is too broad, can we be specific to the module and perhaps consider using `psycopg2.Error`? The exception psycopg2 is likely to go through is https://www.psycopg.org/docs/module.html#exceptions. Could we also be specific about the areas we would want to catch the exception instead of the entire block? For example:

```python
try:
    conn.execute(sql_command, placeholder_values)
except psycopg2.Error